consensus/istanbul/backend/announce.go

```diff
@@ -1006,60 +1046,91 @@ func (sb *Backend) upsertAndGossipVersionCertificateEntries(entries []*vet.Versi
 	return nil
 }
 
-// updateAnnounceVersion will synchronously update the announce version.
+// UpdateAnnounceVersion will synchronously update the announce version.
 // Must be called in a separate goroutine from the announceThread to avoid
 // a deadlock.
-func (sb *Backend) updateAnnounceVersion() {
+func (sb *Backend) UpdateAnnounceVersion() {
 	sb.updateAnnounceVersionCh <- struct{}{}
-	<-sb.updateAnnounceVersionCompleteCh
+}
+
+// GetAnnounceVersion will retrieve the current announce version.
+func (sb *Backend) GetAnnounceVersion() uint {
+	sb.announceVersionMu.RLock()
+	defer sb.announceVersionMu.RUnlock()
+	return sb.announceVersion
 }
 
 // setAndShareUpdatedAnnounceVersion generates announce data structures and
 // and shares them with relevant nodes.
 // It will:
 //  1) Generate a new enode certificate
-//  2) Send the new enode certificate to this node's proxy if one exists
-//  3) Send the new enode certificate to all peers in the validator conn set
-//  4) Generate a new version certificate
-//  5) Gossip the new version certificate to all peers
+//  2) Multicast the new enode certificate to all peers in the validator conn set (note that if this is a proxied validator, it's multicast
+//     message will be wrapped within a forward message to the proxy (which will in turn send the enode certificate to remote validators).
+//  3) Generate a new version certificate
+//  4) Gossip the new version certificate to all peers
 func (sb *Backend) setAndShareUpdatedAnnounceVersion(version uint) error {
```

Comment:
It seems to be assumed that the caller is a validator (and not a proxy) but that is not immediatly clear. Could you add a note, or a check to the start of the function?

Reply:
Ya, it's the validator.  In general, all of the functionality within `announce.go` (which hopefully will be moved to it's own package) is executed only by the validator.